src/native/corehost/hostmisc/utils.cpp

```diff
@@ -98,6 +98,22 @@ pal::string_t strip_file_ext(const pal::string_t& path)
     return path.substr(0, dot_pos);
 }
 
+bool get_line_from_file(FILE* pFile, pal::string_t& line)
+{
+    std::vector<pal::char_t> buffer;
+    char last_char;
+    while ((last_char = fgetc(pFile)))
+    {
+        if (last_char == '\n' || last_char == EOF)
+            break;
+
+        buffer.push_back((pal::char_t)last_char);
+    }
```

Comment:
Perf question - is this "fast enough"?
Alternatively we could call fgets and if the last returned character is not `\n` call it again and concatenate.

I don't know which one is faster (or if the difference is interesting in any way).

Being a bit paranoid here - this is a code path which is taken by every single application, so we need this to be VERY fast.

Reply:
There's a very small win of fgets over the vector approach here, so I changed it.